WordPress/src/main/java/org/wordpress/android/ui/notifications/adapters/NotesAdapter.kt

```diff
@@ -106,34 +84,28 @@ class NotesAdapter(
      */
     fun addAll(notes: List<Note>) = coroutineScope.launch {
         val newNotes = buildFilteredNotesList(notes, currentFilter)
-        val result = DiffUtil.calculateDiff(object : DiffUtil.Callback() {
-            override fun getOldListSize(): Int = filteredNotes.size
-            override fun getNewListSize(): Int = newNotes.size
-            override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean =
-                filteredNotes[oldItemPosition].id == newNotes[newItemPosition].id
-
-            override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean =
-                filteredNotes[oldItemPosition].json.toString() == newNotes[newItemPosition].json.toString()
+        val differ = AsyncListDiffer(this@NotesAdapter, object: ItemCallback<Note>(){
```

Comment:
Replace DiffUtil with AsyncListDiffer because I encountered a IndexOutOfBoundsException, and I am not sure if AsyncListDiffer can fix that problem.